### PR TITLE
Add optional icon and rename component to `GridCellContent`

### PR DESCRIPTION
**Pull request by jamesricky**
---

## PR Checklist

-   [x] Merge dependent PR #2040
-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: SVK-264
-   [x] Provide screenshots/screencasts if the change contains visual changes

### Previously 

Screenshot 2024-05-21 at 15 02 37

### Now

Screenshot 2024-05-21 at 15 02 18

### CODE REVIEW

**Review by johnnyomair — COMMENTED** [review]
Now that we support an icon, is GridCellText still the best name? 😁

**Comment by jamesricky** [comment]
> Now that we support an icon, is GridCellText still the best name? 😁

Maybe `GridCellContent` or just `GridCell`? 
Not sure if the `primary` and `secondary` names are the best names then though 😅

**Comment by nsams** [comment]
> GridCellContent

:+1:

> Not sure if the primary and secondary names are the best names then though

yes, not the best - alternative ideas?

**Comment by johnnyomair** [comment]
> yes, not the best - alternative ideas?

`primaryText` and `secondaryText`

**Comment by johnnyomair** [comment]
@jamesricky please rebase

**Review by nsams — APPROVED** [review]

File: packages/admin/admin/src/dataGrid/GridCellContent.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    const ownerState: OwnerState = {
+        hasSecondaryText: Boolean(secondaryText),
+    };
+
+    return (
+        <Root ownerState={ownerState} {...slotProps?.root} {...restProps}>
+            {icon && <IconContainer {...slotProps?.iconContainer}>{icon}</IconContainer>}
+            <TextContainer>
+                <PrimaryText ownerState={ownerState} {...slotProps?.primaryText}>
+                    {primaryText ? primaryText : children}
```
[the remark this answers is not in the document]
Comment (jamesricky) [reply packages/admin/admin/src/dataGrid/GridCellContent.tsx:42]: As discussed here, we want to have a similar behavior to MUI: https://github.com/vivid-planet/comet/pull/1978#discussion_r1587108951

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**